### PR TITLE
Fixes #13277: Invalid parsing of `--` in CLI

### LINKED ISSUE #13277: Invalid -- parsing in CLI
### What steps will reproduce the problem?
in bash:
`./yii -- help`
### What is the expected result?
Standart yii2 actions list
### What do you get instead?
`Error: Unknown command "--".`

Following the guides, double minus (--) must be parsed as end of any options. More info:
http://www.gnu.org/prep/standards/standards.html#Command_002dLine-Interfaces
http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html (Guideline 10)
### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.8
| PHP version      | 7.0.8
| Operating system | Linux Mint

### PR DESCRIPTION
**Pull request by rugabarbo**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13277

### CODE REVIEW

File: framework/console/Request.php [hunk 1]
```diff
@@ -57,16 +57,27 @@ public function setParams($params)
     public function resolve()
     {
         $rawParams = $this->getParams();
+        $endOfOptionsFound = false;
         if (isset($rawParams[0])) {
             $route = $rawParams[0];
             array_shift($rawParams);
+
+            if ($route == '--') {
```
Comment (samdark) [inline framework/console/Request.php]: `===` can be safely used.

File: framework/console/Request.php [hunk 2]
```diff
@@ -57,16 +57,27 @@ public function setParams($params)
...
         } else {
             $route = '';
         }
 
         $params = [];
         foreach ($rawParams as $param) {
-            if (preg_match('/^--(\w+)(?:=(.*))?$/', $param, $matches)) {
+            if ($endOfOptionsFound) {
+                $params[] = $param;
+            } elseif ($param == '--') {
```
Comment (samdark) [inline framework/console/Request.php]: `===` could be safely used.

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by samdark — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]
Vote for merging in 2.0.11

**Review by dynasource — APPROVED** [review]

**Comment by SilverFire** [comment]
Merged b4acf4606127315e2843e07547b849557705aeec
Thank you! :christmas_tree:

**Pull request closed without merging**